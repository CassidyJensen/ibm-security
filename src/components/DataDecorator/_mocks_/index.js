/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

export default {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
