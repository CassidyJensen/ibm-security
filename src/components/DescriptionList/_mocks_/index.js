/**
 * @file Description list mocks.
 * @copyright IBM Security 2019 - 2020
 */

export default [
  {
    id: 0,
    title: 'Name',
    description: 'Some text entered here',
  },
  {
    id: 1,
    title: 'Longer Name',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc dui magna, finibus id tortor sed, aliquet bibendum augue. Aenean posuere sem vel euismod dignissim. Nulla ut cursus dolor. Pellentesque vulputate nisl a porttitor interdum. ',
  },
  {
    id: 2,
    title: 'Even Longer Name',
    description:
      'Sed ut et perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam.',
  },
];
