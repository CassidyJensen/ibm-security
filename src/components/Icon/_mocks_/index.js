/**
 * @file Icon mocks.
 * @copyright IBM Security 2019
 */

import Add16 from '@carbon/icons-react/lib/add/16';

const path = 'M0';
const renderIcon = Add16;

export { path, renderIcon };
