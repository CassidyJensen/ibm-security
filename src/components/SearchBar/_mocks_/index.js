/**
 * @file Search bar mocks.
 * @copyright IBM Security 2019
 */

const scopes = new Array(3)
  .fill()
  .map((name = 'Scope', id) => ({ id, name: `${name} ${id + 1}` }));

export default {
  submitLabel: 'Submit',
  placeHolderText: 'Placeholder',
  labelText: 'Label',
  clearButtonLabelText: 'Clear',
  value: 'Initial value',
  scopes,
  selectedScopes: [scopes[0]],
  scopesTypeLabel: 'Scopes',
  allSelectedScopesLabel: 'All scopes',
};
