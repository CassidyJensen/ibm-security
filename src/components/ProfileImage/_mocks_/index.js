/**
 * @file Profile image mocks.
 * @copyright IBM Security 2019
 */

import { className } from '../../_mocks_';
import { profile } from '../../Header/_mocks_';

const imageUrl = 'http://via.placeholder.com/160x160';

const profileWithImage = Object.assign({}, profile, {
  image_url: imageUrl,
});

const large = true;

export { className, large, profile, profileWithImage };
