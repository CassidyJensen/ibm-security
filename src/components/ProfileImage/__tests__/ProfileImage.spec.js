/**
 * @file Profile image tests.
 * @copyright IBM Security 2019
 */

import { shallow } from 'enzyme';
import React from 'react';

import { className, large, profile, profileWithImage } from '../_mocks_';

import ProfileImage from '../';

describe('ProfileImage', () => {
  let profileImage;

  beforeEach(() => {
    profileImage = shallow(
      <ProfileImage className={className} profile={profile} />
    );
  });

  it('renders', () => {
    expect(profileImage).toMatchSnapshot();
  });

  it('renders the large variation', () => {
    profileImage.setProps({ large });
    expect(profileImage.render()).toMatchSnapshot();
  });

  it('renders an image', () => {
    profileImage.setProps({
      profile: profileWithImage,
    });

    expect(profileImage).toMatchSnapshot();
  });

  it('renders the large variation with an image', () => {
    profileImage.setProps({
      large,
      profile: profileWithImage,
    });

    expect(profileImage).toMatchSnapshot();
  });
});
