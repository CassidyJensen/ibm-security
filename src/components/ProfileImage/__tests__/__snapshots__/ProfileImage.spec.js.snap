// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage renders 1`] = `
<span
  className="security--profile-image class"
>
  SU
</span>
`;

exports[`ProfileImage renders an image 1`] = `
<img
  alt="Sample User"
  className="security--profile-image class"
  src="http://via.placeholder.com/160x160"
/>
`;

exports[`ProfileImage renders the large variation 1`] = `
<span
  class="security--profile-image class security--profile-image--large"
>
  SU
</span>
`;

exports[`ProfileImage renders the large variation with an image 1`] = `
<img
  alt="Sample User"
  className="security--profile-image class security--profile-image--large"
  src="http://via.placeholder.com/160x160"
/>
`;
